src/metabase/search/filter.clj

```diff
@@ -233,11 +233,18 @@
 
   This is function instead of a def so that optional-filter-clause can be defined anywhere in the codebase."
   []
-  (->> (dissoc (methods build-optional-filter-query) :default)
-       keys
-       (reduce (fn [acc [filter model]]
-                 (update acc filter set/union #{model}))
-               {})))
+  (merge
+   ;; models support search native query is if dataset_query is one of the searchable columns
+   {:search-native-query (->> (dissoc (methods search.config/searchable-columns-for-model) :default)
```

Comment:
Starting to wonder whether this should be a multimethod.

Reply:
wdym by `this`?

Reply:
Sorry, I meant this function. `feature->supported-models`.